Restructure and update contributing guides (#841)
timescale/docs: CONTRIBUTING.md

To contribute to TimescaleDB's source code, see the [TimescaleDB
repository](github.com/timescale/timescaledb).

## Contribute to documentation
You can make contribute to documentation by making a pull request.

### Make a minor change
For minor changes, such as typos and broken-link fixes, you can edit directly in
GitHub. 

#### Making a minor change

1.  Sign in to your [github](github.com) account.
1.  Open the file you want to edit within GitHub. For help finding the right
    file, see the [repository structure section](#repository-structure).
1.  Click the pencil icon in the top-right corner of the code box. GitHub
    automatically forks the project for you and opens a code editor.
1.  Make your edits.
1.  Propose your changes by adding a title and optional description in the
    `Propose changes` box at the bottom of the page.
1.  Review the differences between your changes and the `latest` branch on the
    docs repository. Click `Create pull request`.
1.  Edit the pull request description. Click `Create pull request`. If
    this is your first contribution, you receive a comment asking you to sign
    the Contributor License Agreement.  

> 1.  Sign the agreement so your contribution can be added to the documentation.

Reviewer: Add procedure tags?

the fork to your local machine.

<procedure>

### Forking and cloning the repository
1.  Sign in to your [github](github.com) account.
1.  Navigate to the [Timescale documentation repo](github.com/timescale/docs).
1.  Click the `Fork` button in the top-right corner, and select the account you
    want to use.
1.  Wait for GitHub to create your fork and redirect you.
1.  Clone the repository to your local machine by clicking the green `Code`
    button, copying the HTTPS URL, and running the following command at your
    command prompt:
    ```bash
    git clone <YOUR_FORK_URL>
    ```
1.  List the current remote branches:
    ```bash
    git remote -v
    ```
    This command should list two remotes, both marked `origin`, like this:
    ```bash
    origin  https://github.com/<YOUR_GITHUB_USERNAME>/docs.git (fetch)
    origin  https://github.com/<YOUR_GITHUB_USERNAME>/docs.git (push)
    ```
    The `origin` remotes are your own fork. You can do whatever you want here
    without changing the upstream repository.
1.  Add the docs repo as an upstream:
    ```bash
    git remote add upstream https://github.com/timescale/docs.git
    ```

> 1.  Check that the upstream repo was successfully added:

Reviewer: ```suggestion
1.  Check that the upstream repository is added successfully:
```

    ```bash
    git remote -v
    ```
    This command should now list the two `origin` remotes from before, plus two
    more labelled `upstream`, like this:
    ```bash
    origin  https://github.com/<YOUR_GITHUB_USERNAME>/docs.git (fetch)
    origin  https://github.com/<YOUR_GITHUB_USERNAME>/docs.git (push)
    upstream  https://github.com/timescale/docs.git (fetch)
    upstream  https://github.com/timescale/docs.git (push)
    ```
1.  Fetch the branches in the upstream repository:
    ```bash
    git fetch upstream
    ```
1.  Merge the changes from the upstream `latest` branch into your fork's
    `latest` branch:
    ```bash
    git merge upstream/latest
    ```
1.  Create a new branch for the work you want to do. Give it a descriptive name
    that includes your GitHub username. For example:
    ```bash

>     git checkout -b update-readme-<USERNAME>

Reviewer: I'm reliably informed that we're supposed to use `git switch` instead of `git checkout` these days. Might be worth updating.

Author: I thought `git switch` was still experimental, at least this is what the help page tells me:

`THIS COMMAND IS EXPERIMENTAL. THE BEHAVIOR MAY CHANGE.`

I'm fine with recommending it (it's what I use myself), but do we worry syntax will change?

Reviewer: Ah, that's interesting, I didn't realise GH called it experimental! Let's leave it as-is for now.

    Switched to branch 'latest'
    Your branch is up to date with 'origin/latest'.
    ```
    BEWARE! Despite the message, your branch is probably _not_ up to date. Your
    `origin/` pointers don't reflect the latest changes in the upstream remote
    repository.
1.  Fetch the branches in the upstream repository:
    ```bash
    git fetch upstream
    ```
1.  Merge the changes from the upstream `latest` branch into your fork's
    `latest` branch:
    ```bash
    git merge upstream/latest
    ```
1.  To continue work that you began earlier, check out the branch that contains
    your work. To start new work, create a new branch. 

</procedure>

## Repository structure

## Contributor license agreement

## The Timescale documentation team
*   Ryan Booz <https://github.com/ryanbooz>
*   Lana Brindley <https://github.com/Loquacity>
*   Charis Lam <https://github.com/charislam>

> *   Rajakavitha Kodhandapani <https://github.com/Rajakavitha1>

Reviewer: Add Jacob and Daniel?
